### DISCUSSION TITLE
Issue regarding tailwind 3.0 installation in reactjs

### DISCUSSION THREAD

**Discussion by sayyedfaisal06** [tailwindlabs/tailwindcss#6472] [Help]
As i create a new react project and followed all the steps from tailwind website to add it to my project its not working. Any idea how to do it.

**Comment by NovusOpus** [comment]
Can you be more specific about how it's not working? 

Also, what's the setup you're going for: 

1. Create React App
2. Vite + React
3. Any other setup?

**Reply by BkunS** [reply]
I tried with Create React App according to [official doc](https://tailwindcss.com/docs/guides/create-react-app). It didn't mention adding the `craco.config.js` part and tailwind v3.0 won't work without replacing `react-script` with `craco` (correct me if I'm wrong). After I add `craco.config.js` as following as many have suggested, 
```
module.exports = {
  style: {
    postcss: {
      plugins: [
        require('tailwindcss'),
        require('autoprefixer'),
      ],
    },
  },
}
```
I received this error as turned up in another tailwind discussion [here](https://github.com/tailwindlabs/tailwindcss/discussions/6337):
`Error: PostCSS plugin tailwindcss requires PostCSS 8`

Node: v14.17.0
```
"devDependencies": {
  "autoprefixer": "^10.4.0",
  "postcss": "^8.4.5",
  "tailwindcss": "^3.0.5"
}
```

**Reply by NovusOpus** [reply]
Check this @adamwathan(Tailwind CSS Creator) shared repo: [Create React App Setup with Tailwind CSS V3](https://github.com/adamwathan/cra-tailwind-3), In his [Tweet](https://twitter.com/adamwathan/status/1467300908270862341)

**Reply by BkunS** [reply]
Thx, it did help! I first tried to change my current project to be the same setup as his, but the issue persisted, so I had to re-create the whole project with plain CRA and it worked! I guess CRA with TS template I used before might behave differently underneath.

**Comment by lukewarlow** [comment]
This issue is almost definitely caused by you using Create React App v4 which only supports Postcss 7. You'd need to use the CLI (as suggested above) or use Create React App v5 (alpha) which works with postcss 8.
